projects/packages/connection/src/class-rest-connector.php

```diff
@@ -976,4 +987,37 @@ public static function set_connection_owner_permission_check() {
 
 		return new WP_Error( 'invalid_user_permission_set_connection_owner', self::get_user_permissions_error_msg(), array( 'status' => rest_authorization_required_code() ) );
 	}
+
+	/**
+	 * The endpoint verifies blog connection and blog token validity.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return mixed|null
+	 */
+	public function test_connection() {
+		/**
+		 * Filters the successful response of the XMLRPC test_connection method
+		 *
+		 * @param string $response The response string.
+		 */
+		$status = apply_filters( 'jetpack_xmlrpc_test_connection_response', 'success' );
```

Comment:
Should we create a new hook there, that we'll use instead of the legacy `xmlrpc`?

Reply:
Good point.
I created a new filter without `xmlrpc` in the name and deprecated the old one.

Reply:
Maybe we could keep the old one too, without deprecating it, so that it still applies in cases where we have an older version of the Jetpack plugin, co-existing with the latest version of a standalone that uses the Jetpack Connection package.
For the Jetpack plugin, I guess it would also make sense to add the new filter outside of `class-jetpack-xmlrpc-methods.php`.

Reply:
Adjusted in https://github.com/Automattic/jetpack/pull/36471/commits/76fdf22921cade6e67d5801c694416b1218e483a as per the discussion p1710942827689409-slack-C05PV073SG3